docker-java-transport-okhttp/src/main/java/com/github/dockerjava/okhttp/OkDockerHttpClient.java

```diff
@@ -0,0 +1,284 @@
+                public MediaType contentType() {
+                    return null;
+                }
+
+                @Override
+                public void writeTo(BufferedSink sink) throws IOException {
+                    sink.writeAll(Okio.source(body));
+                }
+            };
+        }
+        switch (request.method()) {
+            case "POST":
+                return RequestBody.create(null, "");
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public Response execute(Request request) {
+        String url = baseUrl.toString();
+        if (url.endsWith("/") && request.path().startsWith("/")) {
+            url = url.substring(0, url.length() - 1);
+        }
+        okhttp3.Request.Builder requestBuilder = new okhttp3.Request.Builder()
+            .url(url + request.path())
+            .tag(Request.class, request)
+            .method(request.method(), toRequestBody(request));
+
+        request.headers().forEach(requestBuilder::header);
+
+        final OkHttpClient clientToUse;
+
+        if (request.hijackedInput() == null) {
+            clientToUse = client;
+        } else {
+            clientToUse = streamingClient;
+        }
+
+        try {
+            return new OkResponse(clientToUse.newCall(requestBuilder.build()).execute());
+        } catch (IOException e) {
+            throw new UncheckedIOException("Error while executing " + request, e);
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        for (OkHttpClient clientToClose : new OkHttpClient[]{client, streamingClient}) {
+            clientToClose.dispatcher().cancelAll();
+            clientToClose.dispatcher().executorService().shutdown();
+            clientToClose.connectionPool().evictAll();
+        }
+    }
+
+    static class OkResponse implements Response {
+
+        static final ThreadLocal<Boolean> CLOSING = ThreadLocal.withInitial(() -> false);
+
+        private final okhttp3.Response response;
+
+        OkResponse(okhttp3.Response response) {
+            this.response = response;
+        }
+
+        @Override
+        public int getStatusCode() {
+            return response.code();
+        }
+
+        @Override
+        public Map<String, List<String>> getHeaders() {
+            return response.headers().toMultimap();
+        }
+
+        @Override
+        public InputStream getBody() {
+            ResponseBody body = response.body();
+            if (body == null) {
+                return null;
+            }
+
+            return body.source().inputStream();
+        }
+
+        @Override
+        public void close() {
+            boolean previous = CLOSING.get();
+            CLOSING.set(true);
+            try {
+                response.close();
+            } finally {
+                CLOSING.set(previous);
+            }
+        }
+    }
+
+    static class TrustAllX509TrustManager implements X509TrustManager {
```

Comment:
Do we really want this? This can lead to MITM attacks on docker credentials (maybe)?

Reply:
reported as https://github.com/docker-java/docker-java/issues/1352, already in master and unrelated to this PR